regular-thread.go

```diff
@@ -0,0 +1,101 @@
+package frankenphp
+
+// #include "frankenphp.h"
+import "C"
+import (
+	"net/http"
+)
+
+// representation of a non-worker PHP thread
+// executes PHP scripts in a web context
+// implements the threadHandler interface
+type regularThread struct {
+	state         *threadState
+	thread        *phpThread
+	activeRequest *http.Request
+}
+
+func convertToRegularThread(thread *phpThread) {
+	thread.handler = &regularThread{
+		thread: thread,
+		state:  thread.state,
+	}
+	thread.state.set(stateActive)
+}
+
+func (t *regularThread) isReadyToTransition() bool {
+	return false
+}
+
+func (handler *regularThread) getActiveRequest() *http.Request {
+	return handler.activeRequest
+}
+
+// return the name of the script or an empty string if no script should be executed
+func (handler *regularThread) beforeScriptExecution() string {
+	currentState := handler.state.get()
+	switch currentState {
+	case stateInactive:
+		handler.state.waitFor(stateActive, stateShuttingDown)
+		return handler.beforeScriptExecution()
+	case stateShuttingDown:
+		return ""
+	case stateReady, stateActive:
+		return handler.waitForScriptExecution()
+	}
+	return ""
+}
+
+// return true if the worker should continue to run
+func (handler *regularThread) afterScriptExecution(exitStatus int) bool {
+	handler.afterRequest(exitStatus)
+
+	currentState := handler.state.get()
+	switch currentState {
+	case stateDrain:
+		return true
+	case stateShuttingDown:
+		return false
+	}
+	return true
```

Comment:
```suggestion
	switch currentState {
	case stateDrain:
		return true
	case stateShuttingDown:
		return false
	default:
		return true
	}
```

Reply:
We can even just keep `stateShuttingDown` and the default case.